starport/services/scaffolder/component.go

```diff
@@ -142,3 +147,72 @@ func isGoReservedWord(name string) bool {
 	}
 	return false
 }
+
+// isComponentCreated checks if the component has been already created with Starport in the project
+func isComponentCreated(appPath, moduleName, compName string) (bool, error) {
+	compNameTitle := strings.Title(compName)
+	typesToCheck := []string{
+		compNameTitle,
+		"MsgCreate" + compNameTitle,
+		"MsgUpdate" + compNameTitle,
+		"MsgDelete" + compNameTitle,
+		"Msg" + compNameTitle,
+		"Query" + compNameTitle + "Request",
+		"Query" + compNameTitle + "Response",
+		"QueryAll" + compNameTitle + "Request",
+		"QueryAll" + compNameTitle + "Response",
+		"QueryGet" + compNameTitle + "Request",
+		"QueryGet" + compNameTitle + "Response",
+		"MsgSend" + compNameTitle,
+		compNameTitle + "PacketData",
+	}
+
+	return checkTypesDefined(appPath, moduleName, typesToCheck)
+}
+
+// checkTypesDefined returns true if at least one of the provided type is defined in the type package of the module
+func checkTypesDefined(appPath, moduleName string, typeNames []string) (exist bool, err error) {
+	if len(typeNames) == 0 {
+		return false, errors.New("no type names provided")
+	}
+
+	absPath, err := filepath.Abs(filepath.Join(appPath, "x", moduleName, "types"))
+	if err != nil {
+		return false, err
+	}
+	fileSet := token.NewFileSet()
+	all, err := parser.ParseDir(fileSet, absPath, func(os.FileInfo) bool { return true }, parser.ParseComments)
+	if err != nil {
+		return false, err
+	}
+
+	for _, pkg := range all {
+		for _, f := range pkg.Files {
+			ast.Inspect(f, func(x ast.Node) bool {
+				// check if it is a type
```

Comment:
not sure if such comments contribute to the clarity of the code, the line below already self-explanatory